R/model.R

```diff
@@ -448,7 +454,9 @@ compile_method <- function(quiet = TRUE,
     }
   }
   stancflags_val <- paste0("STANCFLAGS += ", stancflags_val, paste0(stanc_built_options, collapse = " "))
-  prepare_precompiled(cpp_options, quiet)
+  if (cmdstan_version() < "2.24") {
+    prepare_precompiled(cpp_options, quiet)
```

Comment:
We can leave the `prepare_precompiled` function in for now, but I would maybe remove it after a release cycle, so when 2.25 is out? What do you think?

This function just makes versions 2.23 and earlier a bit more user friendly, but everything still works even without it, though a bit more care needs to be taken when using the thread,mpi and opencl flags with it.

So the function will become more or less dead code once users move to 2.24.

Reply:
Sure, that makes. Let's remove it once 2.25 comes out. Should we open an issue to remind ourselves?